modules/library/main/src/main/java/org/geotools/styling/RasterSymbolizerImpl.java

```diff
@@ -425,10 +422,12 @@ public void setOpacity(Expression opacity) {
      */
     @Deprecated
     public void setOverlap(Expression overlap) {
-        if (this.overlap == overlap) {
+        if(overlap == null)	{
             return;
         }
-        this.overlap = overlap;
+
+        OverlapBehavior overlapBehavior = OverlapBehavior.valueOf(overlap.toString());
```

Comment:
This needs some validation - since Expression can be pretty much anything, and OverlapBehavior only really has 4 valid values, an IllegalArgumentException should be thrown in the case where the Expression is not a valid value